src/main/scala/uk/pkerrigan/darwin/Message.scala

```diff
@@ -0,0 +1,63 @@
+package uk.pkerrigan.darwin
+
+import scala.xml.{Node, XML}
+
+sealed trait Time
+case class EstimatedTime(time: String) extends Time
+case class ActualTime(time: String) extends Time
+
+sealed trait Location {
+  def tiploc: String
+}
+case class Station(tiploc: String, arrivalTime: Time, departureTime: Time) extends Location
+case class Waypoint(tiploc: String, passTime: Time) extends Location
+
+sealed trait Message {
+  def origin: String
+}
+case class ScheduleUpdate(origin: String, runId: String, locations: List[Location]) extends Message
+
+object Message {
+  def parseMessage(message: String): Option[Message] =
```

Comment:
so Eithers look like

```scala
sealed trait Either[A, B]
case class Right[A, B](value: B) extends Either[A, B]
case class Left[A, B](value: A) extends Either[A, B]
```

so like an `Option` but instead of just `None` in the case of failure you have a `Left` with some value in it
conventionally `Left`s are for bad things like errors and `Right`s are for good things

And you can turn an `Option` into an `Either` by calling `.toRight` and supplying a left value to use if the `Option` is `None`

```scala
None.toRight("error of some kind") // Left("error of some kind")
Some(1234).toRight("error of some kind") // Right(1234)
```

so the argument to `toRight` would be some kind of error about the XML